### PR TITLE
Correctly record internal dependencies in compLibDependencies, report…

### PR DESCRIPTION
**Pull request by ezyang**
…ed by @hvr.

Signed-off-by: Edward Z. Yang <ezyang@cs.stanford.edu>

### CODE REVIEW

File: cabal-install/Distribution/Client/ProjectPlanning.hs [hunk 1]
```diff
@@ -1252,7 +1252,8 @@ elaborateInstallPlan verbosity platform compiler compilerprogdb pkgConfigDB
             -- compNonSetupDependencies are defined when we define
             -- 'elab'.
             compLibDependencies =
-                concatMap (elaborateLibSolverId mapDep) external_lib_dep_sids
+                -- concatMap (elaborateLibSolverId mapDep) external_lib_dep_sids
```
Comment (hvr) [inline cabal-install/Distribution/Client/ProjectPlanning.hs]: do you plan to keep that as a comment there?

**Comment by hvr** [comment]
I tried this locally and this fix appears to do what I expect it to; so lgtm!

**Comment by 23Skidoo** [comment]
Do we have a ticket about the problem this is supposed to fix?

**Comment by hvr** [comment]
@23Skidoo not afaik, I was talking to @ezyang on IRC about `plan.json` and wondered why it was missing those intra-pkg refs... :-)

**Comment by ezyang** [comment]
I'm working on an improved version of the patch which tightens up ./Setup configure checks to detect when we aren't feeding enough `--dependency` flags, as was the case here.

**Comment by hvr** [comment]
@ezyang what's the status on this? does it need more testing?

**Comment by ezyang** [comment]
Just needs someone to review the latest patch :)

**Pull request merged**